hs_core/views/resource_folder_hierarchy.py

```diff
@@ -101,7 +101,8 @@ def data_store_structure(request):
     )
 
 
-def data_store_folder_zip(request):
+def data_store_folder_zip(request, res_id=None, input_coll_path=None,
+                          output_zip_file_name=None, remove_original_after_zip=None):
     """

```

Comment:
@hyi The _internal enpoint which invokes data_store_folder_zip is a regular Django view, which uses request.POST. The public endpoint is a Django Rest Framework api_view (you can see the decorator) which uses request.data. I can't edit request.POST directly because it is immutable, so my passthrough function does what it can to marshall the data.